sources/packages/backend/apps/api/src/services/application-offering-change-request/application-offering-change-request.service.ts

```diff
@@ -323,7 +329,21 @@ export class ApplicationOfferingChangeRequestService {
     newRequest.applicationOfferingChangeRequestStatus =
       ApplicationOfferingChangeRequestStatus.InProgressWithStudent;
     newRequest.reason = reason;
-
-    return this.applicationOfferingChangeRequestRepo.save(newRequest);
+    return this.dataSource.transaction(async (transactionEntityManager) => {
+      const applicationOfferingChangeRequest =
```

Comment:
As far as I remember, moving the `save` inside the transaction will not make it part of the transaction.
Did you please confirm if it is working as expected?